File: mobile/smartdesk_mobile/lib/features/tickets/domain/enums.dart
// The ticket enums, kept byte-identical to `SmartDesk.Domain.Enums`.
//
// The API serialises enums as their names, so the wire values here are the C# member
// names. `unknown` exists only so a future backend value can never crash the app.

enum TicketStatus {
  newTicket('New', 'New'),
  assigned('Assigned', 'Assigned'),
  inProgress('InProgress', 'In Progress'),
  onHold('OnHold', 'On Hold'),
  escalated('Escalated', 'Escalated'),
  resolved('Resolved', 'Resolved'),
  closed('Closed', 'Closed'),
  cancelled('Cancelled', 'Cancelled'),
  unknown('Unknown', 'Unknown');

  const TicketStatus(this.wire, this.label);

  /// The exact string the API sends and expects.
  final String wire;

  /// Human wording for the UI ("InProgress" reads badly on a badge).
  final String label;

  static TicketStatus parse(String? value) =>
      TicketStatus.values.firstWhere((s) => s.wire == value, orElse: () => TicketStatus.unknown);

  /// Statuses an employee thinks of as "still being worked on".
  bool get isOpen => this == newTicket || this == assigned || this == inProgress ||
      this == onHold || this == escalated;

  bool get isDone => this == resolved || this == closed;
}

enum TicketPriority {
  low('Low', 'Low'),
  medium('Medium', 'Medium'),
  high('High', 'High'),
  critical('Critical', 'Critical');

  const TicketPriority(this.wire, this.label);

  final String wire;
  final String label;

  static TicketPriority parse(String? value) => TicketPriority.values
      .firstWhere((p) => p.wire == value, orElse: () => TicketPriority.medium);
}

/// Derived SLA position. Computed server-side in C#; the client only renders it.
enum SlaState {
  onTrack('OnTrack', 'On track'),
  atRisk('AtRisk', 'At risk'),
  breached('Breached', 'Breached'),
  notApplicable('NotApplicable', 'Not applicable');

  const SlaState(this.wire, this.label);

  final String wire;
  final String label;

  static SlaState parse(String? value) => SlaState.values
      .firstWhere((s) => s.wire == value, orElse: () => SlaState.notApplicable);
}

/// Lifecycle of one agentic AI workflow run.
enum WorkflowStatus {
  planned('Planned', 'Planning'),
  running('Running', 'Running'),
  awaitingApproval('AwaitingApproval', 'Waiting for manager approval'),
  completed('Completed', 'Completed'),
  failed('Failed', 'Failed'),
  rejected('Rejected', 'Rejected by manager'),
  unknown('Unknown', 'Unknown');

  const WorkflowStatus(this.wire, this.label);

  final String wire;
  final String label;

  static WorkflowStatus parse(String? value) => WorkflowStatus.values
      .firstWhere((s) => s.wire == value, orElse: () => WorkflowStatus.unknown);

  /// While the workflow is still moving, the ticket screen keeps polling for progress.
  bool get isInFlight => this == planned || this == running;
}

/// Per-agent step outcome inside a workflow.
enum AgentStepStatus {
  pending('Pending'),
  running('Running'),
  succeeded('Succeeded'),
  failed('Failed'),
  skipped('Skipped'),
  unknown('Unknown');

  const AgentStepStatus(this.wire);

  final String wire;

  static AgentStepStatus parse(String? value) => AgentStepStatus.values
      .firstWhere((s) => s.wire == value, orElse: () => AgentStepStatus.unknown);
}

enum ApprovalStatus {
  pending('Pending', 'Waiting for manager'),
  approved('Approved', 'Approved'),
  rejected('Rejected', 'Rejected'),
  revisionRequested('RevisionRequested', 'Revision requested'),
  unknown('Unknown', 'Unknown');

  const ApprovalStatus(this.wire, this.label);

  final String wire;
  final String label;

  static ApprovalStatus parse(String? value) => ApprovalStatus.values
      .firstWhere((s) => s.wire == value, orElse: () => ApprovalStatus.unknown);
}

enum RiskLevel {
  low('Low'),
  medium('Medium'),
  high('High');

  const RiskLevel(this.wire);

  final String wire;

  static RiskLevel parse(String? value) =>
      RiskLevel.values.firstWhere((r) => r.wire == value, orElse: () => RiskLevel.low);
}

File: backend/SmartDesk.Domain/Enums/Enums.cs
namespace SmartDesk.Domain.Enums;

/// <summary>Ticket lifecycle states. Transitions are enforced by TicketService, never by the LLM.</summary>
public enum TicketStatus
{
    New = 0,
    Assigned = 1,
    InProgress = 2,
    OnHold = 3,
    Escalated = 4,
    Resolved = 5,
    Closed = 6,
    Cancelled = 7
}

public enum TicketPriority
{
    Low = 0,
    Medium = 1,
    High = 2,
    Critical = 3
}

/// <summary>Where an assignment came from. AiApproved means a manager approved an agent recommendation.</summary>
public enum AssignmentSource
{
    Manual = 0,
    AiApproved = 1
}

/// <summary>How a knowledge article got attached to a ticket.</summary>
public enum ArticleLinkSource
{
    Manual = 0,
    Agent = 1
}

public enum WorkflowStatus
{
    Planned = 0,
    Running = 1,
    AwaitingApproval = 2,
    Completed = 3,
    Failed = 4,
    Rejected = 5
}

public enum AgentStepStatus
{
    Pending = 0,
    Running = 1,
    Succeeded = 2,
    Failed = 3,
    Skipped = 4
}

/// <summary>High-impact actions that must pause for human approval (spec section 9.8).</summary>
public enum ApprovalActionType
{
    Escalate = 0,
    Assign = 1,
    ChangePriority = 2
}

public enum ApprovalStatus
{
    Pending = 0,
    Approved = 1,
    Rejected = 2,
    RevisionRequested = 3
}

public enum RiskLevel
{
    Low = 0,
    Medium = 1,
    High = 2
}

/// <summary>Who performed an audited action.</summary>
public enum ActorType
{
    User = 0,
    System = 1,
    Agent = 2
}

public enum NotificationStatus
{
    Pending = 0,
    Sent = 1,
    Failed = 2
}

/// <summary>Derived SLA position of a ticket. Computed in C#, never by the LLM.</summary>
public enum SlaState
{
    OnTrack = 0,
    AtRisk = 1,
    Breached = 2,
    NotApplicable = 3
}
